Week2/HW02.hs

```diff
@@ -0,0 +1,86 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Week2.HW02 where 
+
+-- Mastermind -----------------------------------------
+
+-- A peg can be one of six colors
+data Peg = Red | Green | Blue | Yellow | Orange | Purple
+         deriving (Show, Eq, Ord)
+
+-- A code is defined to simply be a list of Pegs
+type Code = [Peg]
+
+-- A move is constructed using a Code and two integers; the number of
+-- exact matches and the number of regular matches
+data Move = Move Code Int Int
+          deriving (Show, Eq)
+
+-- List containing all of the different Pegs
+colors :: [Peg]
+colors = [Red, Green, Blue, Yellow, Orange, Purple]
+
+-- Exercise 1 -----------------------------------------
+
+-- Get the number of exact matches between the actual code and the guess
+
+-- Example: exactMatches [Red, Blue, Green, Yellow] [Blue, Green, Yellow, Red] == 0
+-- Example: exactMatches [Red, Blue, Green, Yellow] [Red, Purple, Green, Orange] == 2
+exactMatches :: Code -> Code -> Int
+exactMatches [] [] = 0 
+exactMatches [] _ = 0
+exactMatches _ [] = 0
+exactMatches (c1:xs) (c2:ys) = (if c1 == c2 then 1 else 0) + exactMatches xs ys
```

Comment:
Is there a ternary operator ( a>b ? x: y) equivalent in haskell rather than writing `if-then-else` all the time ?

Reply:
Not exactly a ternery operator but there's a `bool` function.
```haskell
>>> bool x y p == if p then y else x
True
```

In words of hoogle:

> Case analysis for the Bool type. bool x y p evaluates to x when p is False, and evaluates to y when p is True.
> This is equivalent to if p then y else x; that is, one can think of it as an if-then-else construct with its arguments reordered.

Reply:
thanks nitin and jan

i tried this 

> exactMatches a b = sum $ zipWith (\x y -> bool 0 1 (x==y)) a b

but the dot operator as given in the spoiler hint  doesnt work here,
lemme know if i am doing something wrong or if this is how it was supposed to be 

on the same note there are 3 variables passed to `zipWidth` function here so can we use dot operator where more than 1 params are involved ?
or does the dot operator only works where the end/innermost function param is a single value ?